### PR TITLE
setup basic scalaz-http project

### PR DESCRIPTION
**Pull request by prayagupd**
- [ ] setup a very basic scala project with scalafmt etc
- [ ] setup travis but needs to setup CICD credentials

@ehoner

### CODE REVIEW

**Comment by sergei-shabanau** [comment]
We talked about having same build process for all Scalaz projects, like the one used in ZIO project. Should we have it here as well?

**Comment by sergei-shabanau** [comment]
@prayagupd 
Scalaz parsers has the build organized similar to ZIO, originally created by Daniel Li.
May be it would be helpful to look at https://github.com/scalaz/scalaz-parsers
Thanks!

**Comment by prayagupd** [comment]
Thanks @sergei-shabanau for the feedback. I checked scalaz-parser and followed the same scala code formatting and travis. Not sure scalaz-parser is being published from master though. I do not see scalaz-parser artifact in snapshot repo - https://oss.sonatype.org/content/repositories/snapshots/org/scalaz/

**Comment by prayagupd** [comment]
@sergei-shabanau correct there is `if [ "$TRAVIS_PULL_REQUEST" = "false" ]; then sbt publish; fi`, which means every merge on master should have published a snapshot artifact. 

It could be because travis cicd is not setup with credentials. You or someone might have to set it up right?

**Comment by prayagupd** [comment]
Alright, here is [scalaz-parser publish error](https://travis-ci.org/scalaz/scalaz-parsers/builds/420339634#L684), 

```bash
[error] (publish) java.io.IOException: 
Access to URL https://oss.sonatype.org/content/repositories/snapshots/org/scalaz/scalaz-parsers_2.12/0.1.0-SNAPSHOT/scalaz-parsers_2.12-0.1.0-SNAPSHOT.pom 
was refused by the server: Unauthorized
```

**Comment by sergei-shabanau** [comment]
@prayagupd 
Oh, I see, thanks for pointing this out.
Myself, I don't have credentials, but I can ask @ryanonsrc if he has them. Or maybe @danielyli knows.

**Comment by sergei-shabanau** [comment]
I asked about credentials in parsers gitter
https://gitter.im/scalaz/scalaz-parsers
Thanks again @prayagupd !

**Comment by sergei-shabanau** [comment]
@prayagupd @ehoner 
I don't have any response from Dan Li yet.
I checked ZIO project and release is done with this plugin 
https://github.com/olafurpg/sbt-ci-release

It has a good readme that describes how to setup the release process, but there is one special step where Sonatype has to authorize a person to do a release for a domain, `org.scalaz` in our case. For ZIO the person is John and I suppose project lead is the one who can setup the process and provide personal credentials.

**Comment by prayagupd** [comment]
@sergei-shabanau  correct, we are using sbt-ci-release as well - https://github.com/scalaz/scalaz-http/blob/40b27f15bc6b3a7b604a88f31d95a4316c50cc7a/project/plugins.sbt#L3

But release part is missing as right now for missing credentials (same reason as scalaz-parser). But we would have to update `travis.yaml` to use creds following scalaz-zio. Let me know if you get to setup project :)

**Comment by sakshamsharma** [comment]
A minor nit.
I just noticed that the build.sbt imports from the object `Scalaz` (in `project/Scalaz.scala`). Can we change it to `ScalazHTTP` or `ScalazBuild` (like in scalaz-parsers) ? `Scalaz` is a bit too generic.

I'm still travelling, but I can look into setting up the release thing in about a week.

Anything else up for debate in this PR?

**Comment by randbw** [comment]
@sakshamsharma Why is the decision being made to not use the scalaz-sbt plugin?

This PR duplicates a lot of configuration provided by using the scalaz-sbt plugin which was intended to be used to standardise scalaz 8 projects. The plugin handles cross version builds, formatting, compiler warnings, publishing, etc...

[link for reference](https://github.com/scalaz/scalaz-sbt/tree/master/src/main/scala)

File: .scalafmt.conf [hunk 1]
```diff
@@ -21,19 +21,11 @@ spaces {
...
-verticalMultiline {
-  atDefnSite = true
-  arityThreshold = 4
-  newlineAfterOpenParen = true
-  newlineAfterImplicitKW = true
-  newlineBeforeImplicitKW = false
-}
-
 rewrite.rules = [
   AsciiSortImports,
```
Comment (randbw) [inline .scalafmt.conf]: These are also in scalaz-sbt plugin 
https://github.com/scalaz/scalaz-sbt/blob/master/src/main/resources/scalafmt.conf

File: .scalafmt.conf [hunk 2]
```diff
@@ -0,0 +1,31 @@
+align = most
```
Comment (randbw) [inline .scalafmt.conf]: This file already exists in the plugin with the same options: https://github.com/scalaz/scalaz-sbt/blob/master/src/main/resources/scalafmt.conf

**Comment by prayagupd** [comment]
Hi @ehoner @sergei-shabanau @randbw @sakshamsharma do you guys have more feedbacks on this Pull Request?

**Comment by randbw** [comment]
I don't 😃

**Comment by sergei-shabanau** [comment]
@prayagupd 
Looks good to me! Thanks!

I have a related question. I tried to use `scalaz-sbt` plugin in parsers project too. And I'm running into the issue with unresolved scalaz-base snapshot dependency (which previously was ok). Do you have anything extra setup on Travis?

My Travis log is here in case it's helpful 
https://travis-ci.org/scalaz/scalaz-parsers/builds/428407973

**Comment by prayagupd** [comment]
Thank you @sergei-shabanau. You or someone has to merge it at some point then :)

Regarding scalaz-base snapshot, something strange has happened as there are no scalaz artifacts at https://oss.sonatype.org/content/repositories/snapshots/org/scalaz/. I deleted my local artifact `scalaz-base_2.12-8.0.0-SNAPSHOT` and get the same error as you.

**Comment by randbw** [comment]
Perhaps worth mentioning to the scalaz-sbt team?

**Pull request merged**